src/docs/perf/rendering/shader.md

```diff
@@ -0,0 +1,171 @@
+[`transitions_perf_test.dart`]: {{site.github}}/flutter/flutter/blob/master/dev/integration_tests/flutter_gallery/test_driver/transitions_perf_test.dart
+
+## Limitations and considerations
+
+1. **Why not just compile or warm up _all_ possible shaders?**<br>
+   If there are only a limited number of possible shaders,
+   then Flutter could do a warm-up and compile all of them
+   before-hand to avoid such jank.
+   However, for the best overall performance,
+   the Skia GPU backend used by Flutter can dynamically generate
+   shaders based on a lot of parameters at runtime
+   (for example, draws, device models, and driver versions).
+   Due to all possible combinations of those parameters,
+   the number of possible shaders is exponential.
+   In short, Flutter uses programs (app, Flutter, and Skia code)
+   to generate some other programs (shaders). The number of possible
+   shader programs that Flutter can generate is simply too large.
+
+2. **Can SkSLs captured from one device help shader compilation jank
+   on another device?**<br>
+   Theoretically, there's no guarantee that the SkSLs from one device
+   would help on another device (but they also won't cause any troubles
+   if SkSLs aren't compatible across devices).
+   Practically, as shown in the table on this [SkSL-based warmup PR][],
+   SkSLs work surprisingly well
+   even if 1) SkSLs are captured from iOS and then applied to Android devices,
+   or 2) SkSLs are captured from emulators and then applied to real mobile
+   devices. As the Flutter team has only a limited number of devices in the lab,
+   we currently don't have enough data to provide a big picture of cross-device
+   helpfulness. We'd love you to provide us more data points to see how it
+   works in the wild. ([`FrameTiming`][] can be used to compute the worst frame
+   rasterization time in release mode.)
+
+[`FrameTiming`]: {{site.api}}/flutter/dart-ui/FrameTiming-class.html
+[SkSL-based warmup PR]: {{site.github}}/flutter/flutter/issues/53607#issuecomment-608587484
+
+3. **SkSL warm-up doesn't help newer iPhones using Metal.**<br>
+   Flutter recently migrated from OpenGL to Metal for all newer iOS
+   devices. However, Skia currently only implements the SkSL warm-up for
+   OpenGL. So the SkSL warm-up would only speed up older iOS devices
+   by default. If you find shader compilation jank to be an issue
+   for your app on newer iPhones, please let us know by filing a
+   [GitHub issue][]. In the longer term, we have a plan to use test-based
+   shader warm-up to mitigate this. If there's an urgent need for fixing
+   shader compilation jank on newer iPhones, please leave feedback on
+   [Issue 61045][], and we can help you turn on OpenGL for your app.
+
+[GitHub issue]: {{site.github}}/flutter/flutter/issues
+[Issue 61045]: {{site.github}}/flutter/flutter/issues/61045
+
+If you have any questions or suggestions on warm-up in general,
+please comment on [Issue 60313][] and [Issue 53607][].
```

Comment:
Issue 60313 and 53607 are specific to SkSL shader warm-up. For general shader compilation jank (warm-up) discussions, we should direct them to https://github.com/flutter/flutter/issues/32170

Reply:
Done

Reply:
`questions on shader warm-up` -> `questions on SkSL shader warm-up`